build_defs/go.build_defs

```diff
@@ -299,7 +314,7 @@ def go_library(name:str, srcs:list, resources:list=None, asm_srcs:list=None, hdr
     return build_rule(
         name = name,
         srcs = srcs,
-        deps = deps,
+        deps = deps + [pkg_info_rule],
```

Comment:
Hmm, doubt it's much of an issue but we might be able to do something here so we don't build this unless we're trying to build package info files. Maybe if we do `plz build //some/go:library -i go_pkg_info`, it could look for subtargets matching that label.

Reply:
Yeah it sort of has to become a dependency in order for us to discover it later, but I agree it's a bit suboptimal to just build them always.